### ISSUE TITLE
Speech To Text Airpods not used as input source

### ISSUE THREAD

**Issue by donpironet** [watson-developer-cloud/swift-sdk#965]
Hi,

Is it possible that recording via Airpods is not supported? Because it seems that watson is always taking the devices microphone even when the Airpods are connected.

I don't see a way to change this or verify.

Some further info on this usecase:
So we are using the swift SDK and I use the method recognizeMicrophone.

The problem is that the library takes the microphone from the phone instead of the Airpods as input for the voice. So the nurse is standing a couple of meters aways and there is ofcourse a little bit of noise in the room. So watson doesn't understand what the surgeon says. What we want is that I can configure the library to use the bluetooth airpods from Apple als input or that the library detects that there is a bluetooth source as input available that it uses that one that's fine as well.

**Comment by germanattanasio** [comment]
I don't think airpods are supported but the recognize method should support any input and if that's not the case then we should add support for that.

@mkistler do you know swift enough to answer this question?

**Comment by mkistler** [comment]
I don't have any idea why this isn't working for airpods. As you say, I would have expected this to work for whatever device was configured as the input device to the phone.

**Comment by donpironet** [comment]
I've found topics on stackoverflow that for the Airpods to record the options need to be different maybe this can help I have no idea:

https://stackoverflow.com/questions/42862595/airpods-not-working-as-an-input-source-for-voice-recorder-app

**Comment by donpironet** [comment]
Any update on this? Because this is really important for our startup. 
Do we need to remove the swift sdk and make our own audiomanager that sends things over http?

**Comment by donpironet** [comment]
@mkistler any update on this? Because for our company, it is starting to urge...

**Comment by mkistler** [comment]
I just pushed a branch up that adds a `category` parameter to `recognizeMicrophone`. This will allow the user to specify a custom AVAudioSession.Category value to use for the microphone.

From what I could gather from the StackOverflow posts on this problem, I think the category the Swift SDK was using was the problem with getting input from the airpods. With this change, the user can specify a category (it defaults to the previously fixed value used in the SDK) that will work with their particular input device.

For Airpods, I think category to pass to `recognizeMicrophone` is:
```
AVAudioSession.Category.playAndRecord, mode: .default, options: [.defaultToSpeaker, .mixWithOthers, .allowBluetooth]
```

I have not been able to test this yet because I don't currently have a test iOS device handy, so I posted the branch in hopes that someone else can test this out and report the results.

**Comment by donpironet** [comment]
@mkistler thanks for helping. It doesn't work I tested it like this:

`       self.speechToText.recognizeMicrophone(settings: self.recognitionSettings, model: "en-US_BroadbandModel", languageCustomizationID: "", acousticCustomizationID: "", category: AVAudioSession.Category.playAndRecord, callback: callback)`

I don't see a way to set the mode and the options like allowBluetooth?

**Comment by donpironet** [comment]
@mkistler I cloned your branch and added this extra parameter:

`categoryOptions: AVAudioSession.CategoryOptions? = nil,`

and used it like this:

`         try audioSession.setCategory(category ?? AVAudioSession.Category.playAndRecord, mode: .default, options: categoryOptions ?? [.defaultToSpeaker, .mixWithOthers])`

Then I compiled the framework and tested it with the extra options and now it works. 
Can you please add this extra parameter so I don't need to replace the own changed framework with the original one?

**Comment by mkistler** [comment]
Thanks @donpironet .  I see now that it is the options and not the category that we need to expose as a parameter.  I'll update my branch accordingly and let you know when that's ready.

**Comment by mkistler** [comment]
@donpironet I just pushed an updated fix to my `mdk/965-stt-with-airpods` branch.  The new fix adds a `configureSession` parameter:
>  parameter **configureSession**: A Boolean value that specifies whether to configure the AVAudioSession. When `true`, the AVAudioSession is set to a standard configuration for microphone input. When `false`, the current AVAudioSession configuration is used. To use an AVAudioSession configuration other than the standard microphone configuration, set the configuration in your application and specify `false` for the **configureSession** parameter. Default is `true`.

Since the AVAudioSession is a "shared instance", and there are multiple configuration options that can be set on it, I think it makes more sense to let your app configure it the way you want and just tell the SDK to use that configuration, rather than passing in all the various configuration options.

Let me know what you think and whether this solution works for you.

**Comment by donpironet** [comment]
@mkistler thanks. I will test it now. Is it possible that I don't see the changes on the branch, do you still need to push it?

**Comment by mkistler** [comment]
Had a little git hiccup on my end. Should be fixed now.

**Comment by donpironet** [comment]
@mkistler You can merge the branch. It works.

Thanks for the fix!

**Comment by donpironet** [comment]
@mkistler any idea when this branch is going to be merged?

**Comment by mkistler** [comment]
Fix merged to develop in #976 and to master in #977.
